shell/common/resource_cache_limit_calculator.h

```diff
@@ -0,0 +1,46 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_COMMON_RESOURCE_CACHE_LIMIT_CALCULATOR_
+#define FLUTTER_SHELL_COMMON_RESOURCE_CACHE_LIMIT_CALCULATOR_
+
+#include <cstdint>
+#include <unordered_map>
+
+#include "flutter/fml/macros.h"
+#include "flutter/fml/memory/weak_ptr.h"
+
+namespace flutter {
+class ResourceCacheLimitItem {
+ public:
+  // The expected GPU resource cache limit in bytes.
+  virtual size_t GetResourceCacheLimit() = 0;
+
+ protected:
+  virtual ~ResourceCacheLimitItem() = default;
+};
+
+class ResourceCacheLimitCalculator {
+ public:
+  ResourceCacheLimitCalculator(size_t max_bytes_threshold)
+      : max_bytes_threshold_(max_bytes_threshold) {}
+
+  ~ResourceCacheLimitCalculator() = default;
+
+  void AddResourceCacheLimitItem(fml::WeakPtr<ResourceCacheLimitItem> item) {
```

Comment:
Document the threading expecations here, add a `FML_DCHECK` if possible that this is called from the platform thread.

Reply:
documented.  but 'FML_DCHECK' seems that not easy to add.

Reply:
That's fine, I guess we'd have to pass in a tak runner, we'll run into problems anyway if someone uses the weakptr on the wrong thread.

Reply:
Please also add a doc comment to this method about calling it on the platform thread.

Reply:
> Please also add a doc comment to this method about calling it on the platform thread.

done